### DISCUSSION TITLE
Commit Style

### DISCUSSION THREAD

**Discussion by thelostone-mc** [gitcoinco/grants-stack#322] [Q&A] [answered]
Proposal: Follow Conventional Commits style, and always include the github ticket number at the end of the commit message.

for example, we should use this format (from https://github.com/gitcoinco/grants-round/pull/297/commits):

```
chore: redirect to 404 for program/round not found
- add AccessDeniedPage
- set state to check if program / round exists
- set state to check if user has access to program

fixes #243
refs #281
```

On passport, we used @commitlint with these rules: https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional, and enforced with pre-commit git hooks

**Comment by shavinac** [comment]
i like standardizing on the `@commitlint/config-conventional` rules (but i'm biased since we did it on passport :^)

for the issue number, i've seen ppl use `[#999]` in brackets - should we standardize on using phrases like `fixes #999`, `related to #999` ?

**Comment by thelostone-mc** [comment]
I'm a +1 on using @commitlint/config-conventional

For issue numbers I'd say we stick to using `fixes` / `refs` instead of anything else
Essentially we could use this as an example

https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-multi-paragraph-body-and-multiple-footers

**Comment by hansmrtn** [comment]
What should commits that do not relate to fixes or issues be prepended with? Something like `add stuff` or `update stuff`, etc?

**Reply by shavinac** [reply]
they can still follow `@commitlint/config-conventional` style, but without the issue number footer. like
`build: remove unnecessary dependencies` (one line commit msg) or for multiline: 
```
ci: speed up github workflow

- do `tsc` once instead of running it in each step
- something else you did here
```

**Comment by thelostone-mc** [comment, marked as answer]
Plan of action: 
- Intragrate @commitlint/config-conventional as pre-commit hook
- Stick the convention
